task/spd/src/main.rs

```diff
@@ -256,6 +256,9 @@ fn main() -> ! {
         wfi: |notification| {
             let _ = sys_recv_closed(&mut [], notification, TaskId::KERNEL);
         },
+        wfi_or_timeout: |_notification, _timeout| {
```

Comment:
The divergence between I2C controller and target seems significant here, to the point where a new type may be warranted.  My modest proposal:
- Add a new `I2cTarget` vtable type which only includes `enable` and `wfi`
  - Use this type in `operate_as_target`
- Remove `wfi` from the `I2cControl`, so you have to use `wfi_or_timeout`

Reply:
Has a proposal ever been so modest and reasonable?!  Doing that now...

Reply:
Done!